### PR TITLE
Add basic user roles service

### PR DESCRIPTION
**Pull request by DmitriyNoa**
Adds client for the new user roles endpoint https://github.com/lensapp/lenscloud/pull/1315.

### CODE REVIEW

File: src/UserRolesService.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  ["Normal" as TeamEntityKind]: Roles.Member,
+};
+
+class UserRolesService extends Base {
+  async getUserSpaceRole(spaceName: string, userName: string) {
+    const { apiEndpointAddress, fetch } = this.lensPlatformClient;
+
+    const url = `${apiEndpointAddress}/spaces/${spaceName}/users/${userName}/role`;
+
+    const json = await throwExpected(
```
Comment (panuhorsmalahti) [inline src/UserRolesService.ts:33]: This should contain mapping for `404`

File: src/UserRolesService.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      async () => fetch.get(url),
+    );
+
+    const result = (json as unknown) as UserSpaceTeamEntity;
+
+    // TeamEntityKind on client are Roles. Map TeamEntityKind ro Roles.
```
Comment (panuhorsmalahti) [inline src/UserRolesService.ts:39]: typo: ro

File: src/UserRolesService.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  async changeUserSpaceRole(spaceName: string, userName: string, targetRole: Roles): Promise<UserSpaceRole> {
+    const { apiEndpointAddress, fetch } = this.lensPlatformClient;
+
+    const url = `${apiEndpointAddress}/spaces/${spaceName}/users/${userName}/role`;
+
+    // Roles on the API side are TeamEntityKind. Map Roles to TeamEntityKind.
+    const teamRole = rolesToTeamEntityKindMap[targetRole];
+
+    const json = await throwExpected(
+      async () => fetch.patch(url, { role: teamRole }),
```
Comment (panuhorsmalahti) [inline src/UserRolesService.ts:54]: Add mapping for 404 and 422

File: src/UserRolesService.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    const result = (json as unknown) as UserSpaceTeamEntity;
+
+    // TeamEntityKind on client are Roles. Map TeamEntityKind ro Roles.
+    return {
+      role: teamEntityKindToRolesMap[result.role],
+    };
+  }
+
+  async changeUserSpaceRole(spaceName: string, userName: string, targetRole: Roles): Promise<UserSpaceRole> {
```
Comment (panuhorsmalahti) [inline src/UserRolesService.ts:45]: How about call this simply `setUserRole`?

Comment (panuhorsmalahti) [reply src/UserRolesService.ts:45]: `targetRole: Role` could be simply `role: Role`

Comment (panuhorsmalahti) [inline src/UserRolesService.ts:45]: Should we disallow setting role `None`? We could at least do that on the typing level

File: src/UserRolesService.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    // Roles on the API side are TeamEntityKind. Map Roles to TeamEntityKind.
+    const teamRole = rolesToTeamEntityKindMap[targetRole];
+
+    const json = await throwExpected(
+      async () => fetch.patch(url, { role: teamRole }),
+    );
+
+    const result = (json as unknown) as UserSpaceTeamEntity;
+
+    // TeamEntityKind on client are Roles. Map TeamEntityKind ro Roles.
```
Comment (panuhorsmalahti) [inline src/UserRolesService.ts:59]: typo: ro

File: src/UserRolesService.ts (hunk 4 above)
Comment (DmitriyNoa) [reply src/UserRolesService.ts]: Setting to None is handled by the mapping function as setting to Normal/Member.

Comment (panuhorsmalahti) [reply src/UserRolesService.ts]: Yes, but that seems misleading. Shouldn't we only allow setting as Roles.Member? Setting role "None" implies to me that the user should be removed from the Space.

**Review by panuhorsmalahti — APPROVED** [review]

File: src/UserRolesService.ts (hunk 4 above)
Comment (DmitriyNoa) [reply src/UserRolesService.ts]: Added that to type check. We have a check for proper roles values on BE. We can also restrict this on client-side when selecting roles

**Pull request merged**